src/SignalR/src/Microsoft.AspNetCore.SignalR.Common/Protocol/HubMethodInvocationMessage.cs

```diff
@@ -102,6 +132,22 @@ public StreamInvocationMessage(string invocationId, string target, object[] argu
             }
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StreamInvocationMessage"/> class.
+        /// </summary>
+        /// <param name="invocationId">The invocation ID.</param>
+        /// <param name="target">The target method name.</param>
+        /// <param name="arguments">The target method arguments.</param>
+        /// <param name="streams">The target methods stream IDs.</param>
+        public StreamInvocationMessage(string invocationId, string target, object[] arguments, string[] streams)
+            : base(invocationId, target, arguments, streams)
+        {
+            if (string.IsNullOrEmpty(invocationId))
+            {
+                throw new ArgumentNullException(nameof(invocationId));
```

Comment:
Is this argument check the only difference between StreamInvocationMessage and HubMethodInvocationMessage?

Reply:
It seems that the presence of a streams/streamIds property should be enough to distinguish the two types of messages.

Reply:
Also, throwing an `ArgumentNullException` for an empty string isn't right.

Either throw an `ArgumentException` for both, or `ArgumentException` when it is empty.